metals/src/main/scala/scala/meta/internal/metals/Indexer.scala

```diff
@@ -46,35 +46,35 @@ import org.eclipse.lsp4j.Position
  * indexes based on it.
  */
 final case class Indexer(
-    workspaceReload: () => WorkspaceReload,
-    doctor: () => Doctor,
+    workspaceReload: WorkspaceReload,
+    doctor: Doctor,
     languageClient: DelegatingLanguageClient,
-    bspSession: () => Option[BspSession],
+    bspSession: Option[BspSession],
     executionContext: ExecutionContextExecutorService,
     tables: Tables,
-    statusBar: () => StatusBar,
+    statusBar: StatusBar,
     timerProvider: TimerProvider,
-    scalafixProvider: () => ScalafixProvider,
+    scalafixProvider: ScalafixProvider,
     indexingPromise: Promise[Unit],
-    buildData: () => Seq[Indexer.BuildTool],
+    buildData: Seq[Indexer.BuildTool],
     clientConfig: ClientConfiguration,
     definitionIndex: OnDemandSymbolIndex,
-    referencesProvider: () => ReferenceProvider,
-    workspaceSymbols: () => WorkspaceSymbolProvider,
+    referencesProvider: ReferenceProvider,
+    workspaceSymbols: WorkspaceSymbolProvider,
     buildTargets: BuildTargets,
-    interactiveSemanticdbs: () => InteractiveSemanticdbs,
-    buildClient: () => ForwardingMetalsBuildClient,
-    semanticDBIndexer: () => SemanticdbIndexer,
-    treeView: () => TreeViewProvider,
-    worksheetProvider: () => WorksheetProvider,
-    symbolSearch: () => MetalsSymbolSearch,
-    buildTools: () => BuildTools,
-    formattingProvider: () => FormattingProvider,
+    interactiveSemanticdbs: InteractiveSemanticdbs,
+    buildClient: ForwardingMetalsBuildClient,
+    semanticDBIndexer: SemanticdbIndexer,
+    treeView: TreeViewProvider,
+    worksheetProvider: WorksheetProvider,
+    symbolSearch: MetalsSymbolSearch,
+    buildTools: BuildTools,
+    formattingProvider: FormattingProvider,
     fileWatcher: FileWatcher,
-    focusedDocument: () => Option[AbsolutePath],
+    focusedDocument: Option[AbsolutePath],
     focusedDocumentBuildTarget: AtomicReference[b.BuildTargetIdentifier],
     buildTargetClasses: BuildTargetClasses,
-    userConfig: () => UserConfiguration,
+    userConfig: UserConfiguration,
```

Comment:
UserConfig should staty with () since it's a var and might change.

Reply:
@tgodzik should I revert these and start with one by one ?
And how to perform tests for a single change ?

Reply:
If you run `sbt` you should be able to see examples on how to run a single test in the welcome view. You can also looks at the https://scalameta.org/metals/docs/contributors/getting-started site.

I would revert all of the changes and do it one by one or a couple at a time.

Sorry for not responding earlier, but I was off last week and only digging through emails right now.